[GH-1868] Updated documentation to note what auto-reload is disabled when serve() used directly. (#1955)
Kludex/uvicorn: docs/index.md

if __name__ == "__main__":
    asyncio.run(main())
```
    

> When the `uvicorn.Server.serve()` is used directly, the hot reload feature is not available.

Reviewer: ```suggestion
When the `uvicorn.Server()`'s `serve()` and `run()` methods are used directly, the hot reload feature is not available.
```

> When the `uvicorn.Server.serve()` is used directly, the `auto-reload` feature is not available.

Reviewer 2: ```suggestion
When the `uvicorn.Server.serve()` is used directly, the hot reload feature is not available.
```

> This is because the `uvicorn.Server.serve()` function is usually used in production environments where automatic reloading is not desired. In production environments, the application is expected to be stable and changes are typically made through a deployment process rather than by modifying the code directly.

Reviewer: ```suggestion
This is because the `serve()` and `run()` methods functions are usually used in production environments where automatic reloading is not desired. In production environments, the application is expected to be stable and changes are typically made through a deployment process rather than by modifying the code directly.
```
